e2e/tests-dfx/install.bash

```diff
@@ -66,3 +66,15 @@ teardown() {
 
     assert_match "ComputeNetworkNotFound.*nosuch"
 }
+
+@test "install succeeds with arbitrary wasm" {
+    dfx_start
+    dfx canister create --all
+    assert_command dfx canister install e2e_project --wasm "${archive:?}/wallet/0.10.0/wallet.wasm"
```

Comment:
How about calling `dfx canister info` and verifying that the module hash matches the expected value?